### ISSUE TITLE
TypeError - Arg 1 passed to validateAddress() must be an instance of craft\commerce\models\Address?

### ISSUE THREAD

**Issue by Vivid-Marketing** [surprisehighway/craft-avatax#19]
Hi there, 
Recently found a bug in our project I'm hoping you can help me figure out. Launched a site a few months ago using Craft-Avatax with Craft 3.3 and Commerce 2.2.15 - things work great!

I've upgraded our dev and staging environments to Craft 3.4.21 and Commerce to 3.1.5. Existing test accounts in our system can place orders (processed by the Stripe sandbox) perfectly fine. New user accounts however are hitting a TypeError on checkout:

> Argument 1 passed to surprisehighway\avatax\services\SalesTaxService::validateAddress() must be an instance of craft\commerce\models\Address, null given, called in /craft/vendor/surprisehighway/craft-avatax/src/services/SalesTaxService.php on line 553

I had to refactor some of my address input/edit form due to some changes in Commerce , but I haven't touched anything else. 

Any thoughts or ideas would be greatly appreciated. Thank you!

**Comment by siffring** [comment]
@Vivid-Marketing were you able to resolve this? Was it a problem with the plugin?

**Comment by Vivid-Marketing** [comment]
@siffring I got this resolved, yes. Thanks for following up. 

The problem wasn't with the plugin. Long story short, something got wonky in our user address registration process after upgrading Commerce. Come to find out, the users we had in our system prior to the upgrade were successful in transactions because our address form was assigning their address as the primary billing address. 

In looking at the problematic new registrations, addresses were being saved, but not set as the primary by default. Updated our registration process, following Craft Commerce's latest example templates and we seem to be good to go again. 

Sorry for not closing the loop initially. Thanks!
